execution_engine_testing/tests/benches/auction_bench.rs

```diff
@@ -0,0 +1,297 @@
+}
+
+fn setup_bench_run_auction(
+    group: &mut BenchmarkGroup<WallTime>,
+    validator_count: usize,
+    delegator_count: usize,
+) {
+    // Setup delegator public keys
+    let delegator_keys = generate_pks(delegator_count);
+    let validator_keys = generate_pks(validator_count);
+
+    let data_dir = TempDir::new().expect("should create temp dir");
+    let mut builder = run_genesis_and_create_initial_accounts(
+        data_dir.path(),
+        &validator_keys,
+        delegator_keys
+            .iter()
+            .cloned()
+            .map(|pk| pk.to_account_hash())
+            .collect::<Vec<_>>(),
+    );
+
+    let bids: Bids = builder.get_bids();
+    let active_bid = bids.get(&DEFAULT_ACCOUNT_PUBLIC_KEY.clone()).unwrap();
+    assert_eq!(
+        builder.get_purse_balance(*active_bid.bonding_purse()),
+        U512::from(VALIDATOR_BID_AMOUNT),
+    );
+    assert_eq!(
+        *active_bid.delegation_rate(),
+        1,
+        "unexpected delegation rate"
+    );
+
+    let mut next_validator_iter = validator_keys.iter().cycle();
+    for delegator_public_key in delegator_keys {
+        let balance = builder
+            .get_public_key_balance_result(delegator_public_key.clone())
+            .motes()
+            .cloned()
+            .unwrap();
+
+        assert_eq!(U512::from(DELEGATOR_INITIAL_BALANCE), balance);
+
+        let delegation_amount = U512::from(DELEGATION_AMOUNT);
+        let delegator_account_hash = delegator_public_key.to_account_hash();
+        let delegate = {
+            let contract_hash = builder.get_auction_contract_hash();
+            let entry_point = auction::METHOD_DELEGATE;
+            let next_validator_key = next_validator_iter
+                .next()
+                .expect("should produce values forever");
+            let args = runtime_args! {
+                auction::ARG_DELEGATOR => delegator_public_key.clone(),
+                auction::ARG_VALIDATOR => next_validator_key.clone(),
+                auction::ARG_AMOUNT => delegation_amount,
+            };
+            let mut rng = rand::thread_rng();
+            let deploy_hash = rng.gen();
+
+            let deploy = DeployItemBuilder::new()
+                .with_address(delegator_account_hash)
+                .with_stored_session_hash(contract_hash, entry_point, args)
+                .with_empty_payment_bytes(
+                    runtime_args! { ARG_AMOUNT => U512::from(DEFAULT_DELEGATE_COST), },
+                )
+                .with_authorization_keys(&[delegator_account_hash])
+                .with_deploy_hash(deploy_hash)
+                .build();
+
+            ExecuteRequestBuilder::new().push_deploy(deploy)
+        }
+        .build();
+        builder.exec(delegate);
+        builder.expect_success();
+        builder.commit();
+        builder.clear_results();
+    }
+
+    let mut era_end_timestamp = TIMESTAMP_INCREMENT_MILLIS;
+
+    // TODO: use add_bid to add non-genesis validator with stake
```

Comment:
That would be a separate test, correct? If so, let's create a ticket for it and remove the TODO.